dbx.py

```diff
@@ -0,0 +1,125 @@
+import os
+from pyspark.dbutils import DBUtils
+
+MANIFEST_FILE_EXT = 'file_list'
+
+def export_parquet(parquet_path, spark):
+    dbutils = DBUtils(spark)
+    file_list = deep_ls(dbutils, parquet_path, 10)
+    manifest = ''
+    for file in file_list:
+        manifest += file.path + '\n'
+    dbutils.fs.put(parquet_path + '.' + MANIFEST_FILE_EXT, manifest, True)
+
+def get_dataframe_from_parquet(parquet_path, destination, spark):
+    dbutils = DBUtils(spark)
+    files = dbutils.fs.ls(parquet_path + '.' + MANIFEST_FILE_EXT)
+    for file in files:
+        # since its input it has the file and a manifest, we only want the file
+        if not file.name.endswith('.manifest'):
+            manifest_path = file.path
+            break
+    manifest = dbutils.fs.head(manifest_path, 10000).splitlines()
+    for init_file in manifest:
+        path_parts = init_file.split('/')
+        new_rel_path = ''
+        for i, part in enumerate(reversed(path_parts)):
+            if part == 'files':
+                break
+            if part.startswith('part-') and part.endswith('parquet'):
+                part_file = part
+            if i < (len(path_parts) - 1):
+                new_rel_path = os.path.join(part, new_rel_path)
+        copy_dir = os.path.join(destination, new_rel_path.replace('-', '_'))
+        if not file_exists(dbutils, copy_dir):
+            dbutils.fs.mkdirs(copy_dir)
+        # now get the files to put there from our input
+        input_part_path = os.path.join(parquet_path, '../', part_file.replace('-', '_'))
```

Comment:
given the loop above, the part_file variable seems to be undefined half of the time :) I think this deserves splitting to some helper methods it does so many things

Reply:
yeah, it's a beast.  the part_file is the the actual filename of the part (ie `part-00001-whatever.parquet`). the other parts are passed along to construct the path.

but yes, this could be encapsulated and tested, good idea.